File: gfg-weekly-coding-contest-109/Level of Node in Tree.py
#User function Template for python3
class Solution:
    # Version 1: Count
    # TC: O(logn) = O(1), SC: O(1)
    def FindLevel(self,N):
        node = 1
        h = 1
        count = 0
        while count < N:
            count += node
            node = node * (h + 1)
            h += 1
        return h - 1

File: gfg-weekly-coding-contest-109/Level of Node in Tree.cpp

class Solution {
  public:
    // Version 1: Count
    // TC: O(logn) = O(1), SC: O(1)
    long long findLevel(long long N) {
        // code here
        long long cur = 1;
        long long level = 1;
        long long count = 0;
        while (count < N) {
            count += cur;
            cur *= (level + 1);
            level++;
        }
        return level - 1;
    }
};